src/search.jl

```diff
@@ -36,21 +36,30 @@ function Base.searchsorted(a::Range, I::ClosedInterval)
     searchsortedfirst(a, I.left):searchsortedlast(a, I.right)
 end
 
-if VERSION > v"0.5.0-dev+4557"
-    # When running with "--check-bounds=yes" (like on Travis), the bounds-check isn't elided
-    @inline function Base.unsafe_getindex{T}(v::Range{T}, i::Integer)
-        convert(T, first(v) + (i-1)*step(v))
-    end
+# When running with "--check-bounds=yes" (like on Travis), the bounds-check isn't elided
+@inline function Base.unsafe_getindex{T}(v::Range{T}, i::Integer)
```

Comment:
I'd rather have `unsafe_getindex` be removed completely and use a new name for the range computation since it's not really unsafe (in terms of memory or validity) at all.  For this use-case we really don't want this function to apply at all to non-ranges.  But this really isn't a high priority for me — `unsafe_getindex` is an undocumented function that's rarely used these days.